plugins/topic.py

```
@@ -32,18 +32,14 @@ def make_topic(self, msg, new_topic=None, new_music=None):
             )
             self.bot.set_topic(topic_string, msg.chan)
 
-    async def find_title(self, url):
-        async with aiohttp.ClientSession() as session:
-            async with session.get(url) as r:
-                page_bytes = await r.read()
-
-        page = page_bytes.decode("utf-8")
-        title_start = page.find("<title>")
-        title_end = page.find("</title>")
-        if title_start > 0 and title_end > 0:
-            title_start += len("<title>")
-            return page[title_start:title_end]
-        else:
+    def find_title(self, url):
+        ydl_opts = {}
+        try:
+            with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+                info_dict = ydl.extract_info(url, download=False)
+                video_title = info_dict.get("title", None)
+            return video_title
+        except Exception:
```

Qu'est-ce que tu cherches à chopper comme exception ici ? utiliser `Exception` c'est presque aussi moche que utiliser un bare except

---

Je cherche à chopper toutes les exceptions. En fait, j'ai déjà vu par le passé que youtube n'aime pas trop qu'on joue avec leur site. Du coup, de temps en temps, youtube-dl casse parce qu'ils ont fait un changement. Donc que ce soit un site non supporté ou une version à upgrade, l'idée est de juste pas planter LechBot

---

C'est déjà ce que fait cette ligne non ? https://github.com/UrLab/lechbot/pull/64/files/a98fb38515870f4805a45117b5a0b7570cac5d94#diff-d47b5ffeb36b68aba9f0bc8651860edeb4beda6eb858dcab4350849b3743655aR75
En plus, la L75 log l'erreur, ça semble plus propre

---

Tu as raison. Je viens de push un correctif là dessus